docs/node/run-your-node/validator-node.mdx

```diff
@@ -330,90 +214,132 @@ To use this configuration, save it in the `/serverdir/etc/config.yml` file and p
   #
-  # file: /var/log/oasis-node.log
+  # log:
+  #   level: debug
+  log:
+    format: JSON
+    level:
+      # Per-module log levels. Longest prefix match will be taken. Fallback to
+      # "default", if no match.
+      default: debug
+      cometbft: warn
+      cometbft/context: error
+    # By default logs are output to stdout. If you would like to output logs to
+    # a file, you can use:
+    #
+    #  file: /var/log/oasis-node.log
+
+# Consensus backend.
+consensus:
+  # CometBFT backend configuration.
+  listen_address: tcp://0.0.0.0:26656
+
+  # The external IP that is used when registering this node to the network.
+  # NOTE: If you are using the Sentry node setup, this option should be
+  # omitted.
+  external_address: tcp://{{ external_address }}:26656
 
 # Genesis.
 genesis:
   # Path to the genesis file for the current version of the network.
   file: /serverdir/etc/genesis.json
 
-# Worker configuration.
-worker:
+# Peer-to-peer.
+p2p:
+  port: 9200
   registration:
-    # In order for the node to register itself, the entity.json of the entity
-    # used to provision the node must be available on the node.
-    entity: /serverdir/node/entity/entity.json
+    addresses:
+      - "{{ external_address }}:9200"
+  # List of seed nodes to connect to.
+  # NOTE: You can add additional seed nodes to this list if you want.
+  seeds:
+    - "{{ seed_node_address }}"
+
+# In order for the node to register itself, the entity.json of the entity
+# used to provision the node must be available on the node.
+registration:
+  entity: /serverdir/node/entity/entity.json
+```
 
-# Consensus backend.
-consensus:
-  # Setting this to true will mean that the node you're deploying will attempt
-  # to register as a validator.
-  validator: true
-
-  # Tendermint backend configuration.
-  tendermint:
-    core:
-      listen_address: tcp://0.0.0.0:26656
-
-      # The external IP that is used when registering this node to the network.
-      # NOTE: If you are using the Sentry node setup, this option should be
-      # omitted.
-      external_address: tcp://{{ external_address }}:26656
-
-    # List of seed nodes to connect to.
-    # NOTE: You can add additional seed nodes to this list if you want.
-    p2p:
-      seed:
-        - "{{ seed_node_address }}"
+#### Starting Oasis Node for the First Time
+
+Execute the Oasis node and pass the config file above by invoking:
+
+```shell
+oasis-node --config /serverdir/etc/config.yml
 ```
 
+After the node starts, it will automatically register itself to the peer-to-peer
+network. The following files will be generated in your `/serverdir/node` folder:
+
+* `consensus.pem`: The node's consensus private key. **DO NOT SHARE**
+* `consensus_pub.pem`: The node's consensus public key.
+* `identity.pem`: The node's identity private key. **DO NOT SHARE**
+* `identity_pub.pem`: The node's identity public key.
+* `node_genesis.json`: The node's details if you wish to include this node in
```

Comment:
I believe this is not generated in this case and should be removed. It was only generated when an explicit `node init` was used before.

Reply:
Running `oasis-node --config config.yml` actually generates those files on my machine in the data dir. Shouldn't it? I'm using version 23.0.4.

Reply:
If I remember correctly, in all versions os Oasis Node these files were generated if they did not exist.

Reply:
I was referring specifically to `node_genesis.json`, not the other key files.